components/linkedin/actions/create-text-post-user/create-text-post-user.mjs

```diff
@@ -0,0 +1,51 @@
+import axios from "axios";
+
+export default {
+  key: "linkedin_create-text-post-user",
+  name: "Create a Text Post (User)",
+  description: "Create shares on LinkedIn using text, URLs, and images. [See the docs](https://docs.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/share-on-linkedin#creating-a-share-on-linkedin) for more information",
+  version: "0.0.11",
+  type: "action",
+  props: {
+    name: {
+      type: "string",
+      label: "Text",
+      description: "Text to be posted on LinkedIn timeline",
+    },
+    linkedin: {
+      label: "LinkedIn",
+      description: "LinkedIn Integration",
+      type: "app",
+      app: "linkedin",
+    },
+  },
+  async run({ $ }) {
+    const data = {
+      "author": `urn:li:person:${this.linkedin.$auth.oauth_uid}`,
+      "lifecycleState": "PUBLISHED",
+      "specificContent": {
+        "com.linkedin.ugc.ShareContent": {
+          "shareCommentary": {
+            "text": this.name,
+          },
+          "shareMediaCategory": "NONE",
+        },
+      },
+      "visibility": {
+        "com.linkedin.ugc.MemberNetworkVisibility": "PUBLIC",
+      },
+    };
+    try {
+      await axios.post("https://api.linkedin.com/v2/ugcPosts", data, {
+        headers: {
+          "Authorization": `Bearer ${this.linkedin.$auth.oauth_access_token}`,
+          "X-Restli-Protocol-Version": "2.0.0",
+        },
+      });
+      $.export("$summary", "Successfully created a new Text Post as User");
```

Comment:
I would store the create post response and also `return` that after the `$summary` export, so the user has access to e.g. the post ID and other things that they can use in the rest of their workflow.

Reply:
implemented, now the $return_value is "id": "urn:li:share:XXX"